src/components/vega-lite/index.tsx

```diff
@@ -3,16 +3,19 @@ import * as vega from 'vega';
 import * as vl from 'vega-lite';
 import {TopLevelExtendedSpec} from 'vega-lite/build/src/spec';
 import * as vegaTooltip from 'vega-tooltip';
+import {LOG_ERROR_CHANGE, LOG_WARNINGS_ADD, LogAction} from '../../actions/log';
+import {ActionHandler} from '../../actions/redux-action';
+import {localLogger} from '../../store/index';
 
-export interface VegaLiteProps {
+export interface VegaLiteProps extends ActionHandler<LogAction> {
```

Comment:
Update: I would prefer having ~~`warn` and `error` methods as~~
a logger as `VegaLiteProps` and let each logger in `Plot` calls handleAction instead.  (You can name it `plotLogger`)